### PR TITLE
COMMERCE-10726 Technical task COMMERCE-9385 Drop DDM Form for product options (3/4)

### PR DESCRIPTION
**Pull request by ghost**
https://issues.liferay.com/browse/COMMERCE-10726

https://github.com/brianchandotcom/liferay-portal/pull/133534

@alessiorendina

### CODE REVIEW

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/rest-openapi.yaml [hunk 1]
```diff
@@ -691,18 +691,42 @@ components:
...
+                    type: boolean
+                priceContributor:
+                    example: true
+                    type: boolean
+                required:
+                    example: true
+                    type: boolean
+                skuContributor:
+                    example: true
+                    type: boolean
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/rest-openapi.yaml:717]: I don't think you should add these fields to this entity

**Review by andrea-ale-sbarra — COMMENTED** [review]
The fields belong to ProducOption not to SkuOption, why did you add them to this entity?

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/util/v1_0/SkuOptionUtil.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+						id = cpDefinitionOptionRel.getCPDefinitionOptionRelId();
+						key = cpDefinitionOptionRel.getKey();
+						skuOptionValueId =
+							cpDefinitionOptionValueRel.
+								getCPDefinitionOptionValueRelId();
+						skuOptionValueKey = cpDefinitionOptionValueRel.getKey();
+						value = String.valueOf(
+							cpDefinitionOptionValueRel.
+								getCPDefinitionOptionValueRelId());
+					}
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/util/v1_0/SkuOptionUtil.java:60]:
```suggestion
                                                 id = String.valueOf(
							cpDefinitionOptionRel.
								getCPDefinitionOptionRelId());
						key = cpDefinitionOptionRel.getKey();
						skuOptionId = cpDefinitionOptionRel.getCPDefinitionOptionRelId();
						skuOptionValueId =
							cpDefinitionOptionValueRel.
								getCPDefinitionOptionValueRelId();
						skuOptionValueKey = cpDefinitionOptionValueRel.getKey();
						value = String.valueOf(
							cpDefinitionOptionValueRel.
								getCPDefinitionOptionValueRelId());
					}
```

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/util/v1_0/SkuOptionUtil.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+				SkuOption skuOption = new SkuOption() {
+					{
+						id = cpDefinitionOptionRel.getCPDefinitionOptionRelId();
+						key = cpDefinitionOptionRel.getKey();
+						skuOptionValueId =
+							cpDefinitionOptionValueRel.
+								getCPDefinitionOptionValueRelId();
+						skuOptionValueKey = cpDefinitionOptionValueRel.getKey();
+						value = String.valueOf(
+							cpDefinitionOptionValueRel.
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/util/v1_0/SkuOptionUtil.java:58]:
```suggestion
						id = cpDefinitionOptionRel.getCPDefinitionOptionRelId();
						key = cpDefinitionOptionRel.getKey();
						skuOptionValueKey = cpDefinitionOptionValueRel.getKey();
						value = String.valueOf(
							cpDefinitionOptionValueRel.
						valueId =
							cpDefinitionOptionValueRel.
								getCPDefinitionOptionValueRelId();
```

Comment (andrea-ale-sbarra) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/util/v1_0/SkuOptionUtil.java:58]: In my opinion one of these two for consistency, plus I don't understand why you need to also add the toString of the value = String.valueOf(cpDefinitionOptionValueRel.) and if you need this why you didn't add the toString of the optionId?

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/commerce-product-api/src/main/java/com/liferay/commerce/product/catalog/CPCatalogEntry.java [hunk 4]
```diff
@@ -24,6 +26,8 @@ public interface CPCatalogEntry {
 
 	public long getCPDefinitionId();
 
+	public List<CPDefinitionOptionRel> getCPDefinitionOptionRels();
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-api/src/main/java/com/liferay/commerce/product/catalog/CPCatalogEntry.java:29]: why did you add this? it look it's never used

Comment (CrescenzoRegaLF) [reply modules/apps/commerce/commerce-product-api/src/main/java/com/liferay/commerce/product/catalog/CPCatalogEntry.java:29]: it is an acceptance criteria to create a getCPDefinitionOptionRels method in the CPCatalogEntry class

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/util/v1_0/SkuOptionUtil.java (hunk 3 above)
Comment (CrescenzoRegaLF) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/util/v1_0/SkuOptionUtil.java:58]: Before the changes, the value field was of integer type and was set to `cpDefinitionOptionValueRel.getCPDefinitionOptionValueRelId()` we changed it to string type but to maintain backwards compatibility we must continue to set the field to the value of `cpDefinitionOptionValueRel.getCPDefinitionOptionValueRelId()`

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Review by andrea-ale-sbarra — APPROVED** [review]
LGTM.

Thank you!

**Comment by andrea-ale-sbarra** [comment]
ci:reevaluate:1377401_11165

**Comment by andrea-ale-sbarra** [comment]
ci:reevaluate:1377401_11165

**Comment by baharturk** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by baharturk** [comment]
ci:test:relevant

**Comment by baharturk** [comment]
ci:test:relevant

**Comment by baharturk** [comment]
ci:test:relevant

**Comment by baharturk** [comment]
ci:forward

**Comment by baharturk** [comment]
ci:forward

**Comment by baharturk** [comment]
ci:forward
